processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -135,12 +178,15 @@ public ExtractionFn getExtractionFn()
 
     byte[] extractionFnBytes = extractionFn == null ? new byte[0] : extractionFn.getCacheKey();
 
+    byte[] orderingBytes = StringUtils.toUtf8(ordering);
+
     ByteBuffer boundCacheBuffer = ByteBuffer.allocate(
-        9
+        10
         + dimensionBytes.length
         + upperBytes.length
         + lowerBytes.length
         + extractionFnBytes.length
+        + orderingBytes.length
     );
     boundCacheBuffer.put(DimFilterUtils.BOUND_CACHE_ID)
                     .put(boundType)
```

Comment:
Might as well remove the AlphaNumericByte for cache key since it's not used for equal or hashCode.